src/rules/extensions.js

```diff
@@ -37,15 +39,15 @@ module.exports = function (context) {
     const extension = path.extname(resolvedPath || importPath).substring(1)
 
     if (!extension || !endsWith(importPath, extension)) {
-      if (isUseOfExtensionEnforced(extension)) {
+      if (isUseOfExtensionRequired(extension) && !isUseOfExtensionForbidden(extension)) {
```

Comment:
so, the reason bracket access and `in` won't work is because someone could define an extension named "toString" or "hasOwnProperty" or similar. The reason to avoid `Object.prototype.hasOwnProperty.call` is because that relies on both `Function#call` and `Object#hasOwnProperty` being undisturbed in the environment. Using `has` is robust against both.